osu.Game/Screens/OnlinePlay/Playlists/PlaylistsReadyButton.cs

```diff
@@ -80,9 +86,16 @@ public override LocalisableString TooltipText
             }
         }
 
-        private bool enoughTimeLeft =>
+        private bool enoughTimeLeft()
+        {
+            // this doesn't consider mods which apply variable rates, yet.
+            double rate = ModUtils.CalculateRateWithMods(mods.Value);
+
+            double hitLength = Math.Round(gameBeatmap.Value.Track.Length / rate);
+
             // This should probably consider the length of the currently selected item, rather than a constant 30 seconds.
-            room.EndDate != null && DateTimeOffset.UtcNow.AddSeconds(30).AddMilliseconds(gameBeatmap.Value.Track.Length) < room.EndDate;
+            return room.EndDate != null && DateTimeOffset.UtcNow.AddSeconds(30).AddMilliseconds(hitLength) < room.EndDate;
```

Comment:
I'm not sure regarding the true purpose of the 30 second addition (maybe to account for possible delays during gameplay start and score submission?), but it does _not_ seem to be a replacement for "length of the currently selected item" (which I assume the track length would be).

While at it, should we update the comment here to make a bit more sense?

Reply:
The comment originiated from this commit https://github.com/ppy/osu/pull/15307/commits/1c578f285accf3f17168708ba5c6aa1fd402f6ec. I'm not sure what the comment meant myself and it sounds like it can be removed if the playlist item length is appropriately accounted for now.

That said, we can definitely take another step forward on computing the length by using `CalculatePlayableLength()` or just `BeatmapInfo.Length` similar to song select:
https://github.com/ppy/osu/blob/c37f617e1d620e726268f38fce59d4c0cd9dd866/osu.Game/Screens/Select/BeatmapInfoWedge.cs#L423

Reply:
That's precisely where the approach (and the comment I pointed out in the first message) is from, keeping `Track.Length` was a conscious choice so far, in case there are any non-skip enjoyers around. Granted, this makes the `hitLength` name somewhat inaccurate.

Reply:
Rename it and add a comment.

Reply:
That said, I would personally prefer using the smallest timespan that is sensible (i.e. only playable time and without the 30 second offset), so if you think that this is the better approach then I'll happily change it.

Reply:
The button is probably meant to be disabled so the user does not submit a score beyond the end time, if there are non-playable boundaries before or after the beatmap's bounds and the user thinks it's fine to wait on them, then the maximum we should allow is the track's full length, anything beyond that is up to the user. I don't think it's good behaviour for the user to not be able to submit a score because they didn't skip.

Reply:
Should be resolved, thanks for the input!